Snakefile.gsc.smk

```diff
@@ -17,45 +28,44 @@ else:
         biopsy: eg biop2 - optional - find libraries
         tumour_lib:
         normal_lib:
-        create_config: (optional) specify output config filename
+        gsc_config_filename: (optional) specify output config filename
         project: (optional) specify project
     """)
     sys.exit()
 
-scripts_dir = os.path.join(workflow.basedir, "scripts")
-sys.path.append(scripts_dir)
-from gsc_build_config import CONFIG_BASENAME, get_biopsy_dna_tumour_normal, get_gsc_output_folder
-
 if "biopsy" in config.keys():
-    tumour_lib, normal_lib = get_biopsy_dna_tumour_normal(patient_id=config["id"], biopsy=config["biopsy"])
-if "tumour_lib" in config.keys():
-    tumour_lib = config["tumour_lib"]
-if "normal_lib" in config.keys():
-    normal_lib = config["normal_lib"]
+    config["tumour_lib"], config["normal_lib"] = get_biopsy_dna_tumour_normal(patient_id=config["id"], biopsy=config["biopsy"])
 
 # Find an output filename for the config we are generating
-if "create_config" in config.keys():
-    output_filename = config["create_config"]
+if "gsc_config_filename" in config.keys():
+    output_filename = config["gsc_config_filename"]
 else:
-    output_filename = get_gsc_output_folder(
+    output_dir = get_gsc_output_folder(
         patient_id=config["id"],
-        tumour_lib=tumour_lib,
-        normal_lib=normal_lib,
+        tumour_lib=config["tumour_lib"],
+        normal_lib=config["normal_lib"],
         pipeline_ver=pipeline_ver,
         ploidetect_ver=ploidetect_ver,
         project=config["project"] if "project" in config.keys() else None,
     )
-    output_filename = os.path.join(output_filename, CONFIG_BASENAME)
+    output_filename = os.path.join(output_dir, CONFIG_BASENAME)
+
+# Script building the config if needed
+if not os.path.exists(output_filename):
+    logger.info(f"Creating {output_filename}")
```

Comment:
the logger seems to result in duplication of the stdout to stderr, which ends up being very verbose when doing things like dryruns. Is it necessary over a print statement?

Reply:
Converting to a print statement.

In python scripts using the logging is often better, because debug, warning, errors can be separated. 

Snakemake seems to be doing something special with these though.